src/pages/en/guides/deploy/github.md

```diff
@@ -15,15 +15,27 @@ Astro maintains the official `withastro/action` to deploy your project with very
 
 1. Set the [`site`](/en/reference/configuration-reference/#site) and, if needed, [`base`](/en/reference/configuration-reference/#base) options in `astro.config.mjs`.
     - `site` should be something like `https://<YOUR_USERNAME>.github.io`
-    - `base` should be your repository’s name starting with a forward slash, for example `/my-repo`.
+    - `base` should be your repository’s name starting with a forward slash, for example `/my-repo`. This is so that Astro understands your website's root is `/my-repo`, rather than the default `/`.
     
     :::note
-    If your repository is named `<YOUR_USERNAME>.github.io`, you don’t need to include `base`.
+    Don't set a `base` parameter if:
+
+    - your repository is named `<YOUR_USERNAME>.github.io`
+    - you're using a custom domain
+
+    :::
+
+    :::caution
+    If you are setting up `base` only to deploy to GitHub do not forget to update your routes to include your `base`.
```

Comment:
```suggestion
        If you did not previously have a value for `base` set, and are only configuring this value so that you can deploy to GitHub, you must update your internal page links to now include your `base`.
```
What do you (all!) think about something more "positively" phrased (you must do this), rather than "Don't forget  to!" which may or may not have been clear to the reader before?